File: 167. Two Sum II - Input Array Is Sorted/main.py
class Solution:
    def twoSum(self, numbers: list[int], target: int) -> list[int]:
        l, r = 0, len(numbers) - 1

        while l < r:
            sum = numbers[l] + numbers[r]
            if sum == target:
                return [l + 1, r + 1]
            elif sum < target:
                l += 1
            else:
                r -= 1
        

File: 167. Two Sum II - Input Array Is Sorted/main.js
class Solution {
  twoSum(numbers, target) {
    let l = 0,
      r = numbers.length - 1;

    while (l < r) {
      const sum = numbers[l] + numbers[r];
      if (sum === target) {
        return [l + 1, r + 1];
      } else if (sum < target) {
        l += 1;
      } else {
        r -= 1;
      }
    }
  }
}
